### PR TITLE
Added Non-Functional Changes

### PR DESCRIPTION
**Pull request by yogendra**
Added testing support and test case for generate.js
Added editor config, eslint and editorconfig

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 node_modules
 npm-debug.log
-
+coverage
```
Comment (vot) [inline .gitignore]: Could you potentially move the relevant contents of `.npmignore` to `.gitignore`? npm by default inherits contents of `.gitignore` and it's easier to maintain a single list where possible.

I think `.vscode`, `coverage` and `yarn.lock` could be placed in `.gitignore`. `.npmignore` could go then.
I see no harm in having tests and lint configuration in the package but definitely we need readme. I'd also rather not add editor-specific configurations to the repo.

File: package.json [hunk 2]
```diff
@@ -17,10 +17,7 @@
     "xkp": "cli.js",
     "xkpasswd": "cli.js"
   },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/vot/xkpasswd-node.git"
-  },
+  "repository": "https://github.com/vot/xkpasswd-node.git",
```
Comment (vot) [inline package.json]: Never paid much attention to the format of this in other `package.json` files. Is this a valid shorthand?

I'm looking at https://docs.npmjs.com/getting-started/using-a-package.json where they simply use the old format so I'd rather stick to that but I'm curious, is the one-line repository a thing now?

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+    "extends": "eslint:recommended",
+    "root": true,
+    "env": {
+        "browser": true,
+        "commonjs": true,
+        "es6": true,
+        "node": true
```
Comment (vot) [inline .eslintrc.json]: I haven't checked out your branch yet to test it myself but are these settings correct? I think only `node` should be set to true here, no ES6 features are used that I can think of and it wouldn't run in a browser as-is.

**Comment by vot** [comment]
Hey buddy,
This is really amazing, thanks for getting this module tested and linted.
I've added a few comments in the annotations, let me know if they sound reasonable.

Thank you for putting time into this :)

**Comment by yogendra** [comment]
@vot , I can't see the annotations in the files. Can you tell me which commit/file is that?

**Comment by vot** [comment]
@yogendra - I just realised I had to submit the review in order for you to see it. You should now see the comments.

File: .eslintrc.json (hunk 3 above)
Comment (yogendra) [reply .eslintrc.json]: Agree.

File: .gitignore (hunk 1 above)
Comment (yogendra) [reply .gitignore]: I like to maintain single ignore file too. 
But, as per [this](https://yarnpkg.com/en/docs/yarn-lock#toc-check-into-source-control) `yarn.lock` should be checked in. This ensures that anyone running yarn will get exact set of transitive dependencies (like npm shrinkwrap). 
`.vscode` is ide configuration for debugging, so I thought it would make it handy to quickly debug test/source. 
but neither of these needs to be published on NPM, so you will still have to maintain `.npmignore`

File: package.json (hunk 2 above)
Comment (yogendra) [reply package.json]: i think this came up in some linter (may be), hence I split it. But I am okay both ways.

File: .gitignore (hunk 1 above)
Comment (yogendra) [reply .gitignore]: Also, I thought we can reduce module size by only including required files and no tests. That's why I split gitignore and npmignore

Comment (vot) [reply .gitignore]: I don't think there's need for that level of shrink-wrapping - the only module currently required is `clarg` which is another thing I maintain and it has no dependencies of its own either (that's the entire point of it actually - it's a simple CLI argument parser).

While it could potentially affect tests I don't think that should be enough reason to add it to the repo quite honestly. I'm really inclined to just carry on with the simpler approach of single ignore file (as for tests I think there's no harm in including them in the actual package).

File: package.json (hunk 2 above)
Comment (vot) [reply package.json]: Hmm cool. I'd rather just go with the documented npm way, just for the sake of ensuring it doesn't cause any weird `package.json` issues in older versions or private/cache servers.

Comment (yogendra) [inline package.json]: Yeah that does make sense. So only corner case is skipping `test` directory from publishing? I agree its not a lot of extra files, so should ok to keep it.

Comment (yogendra) [inline package.json]: Ah didn't see the last line on last comment. I think this is resolved. Let me make these changes.

Comment (yogendra) [reply package.json]: Agree.

File: .gitignore (hunk 1 above)
Comment (yogendra) [reply .gitignore]: Ok sold!

**Comment by vot** [comment]
Thanks @yogendra for making the changes, merging it now.
